pydantic/_internal/_fields.py

```diff
@@ -121,8 +121,15 @@ def collect_model_fields(  # noqa: C901
             # Note: we may need to change this logic if/when we introduce a `BareModel` class with no
             # protected namespaces (where `model_config` might be allowed as a field name)
             continue
+
         for protected_namespace in config_wrapper.protected_namespaces:
-            if ann_name.startswith(protected_namespace):
+            ns_violation: bool = False
+            if isinstance(protected_namespace, Pattern):
+                ns_violation = protected_namespace.match(ann_name) is not None
```

Comment:
Is `match` better than `fullmatch` here? I think `match` might be equally capable and more practical (don't need to include `.*` on the end to serve as a prefix match I guess), just want to make sure you've thought about it and decided to go with `match`.

Reply:
Good question - I think `match` is more practical, as you can use special characters (ex: `^<pattern>$`) to basically ensure `fullmatch` behavior, whereas `fullmatch` does that by default and thus offers less customizability 👍